src/plugins/intel_gna/src/transformations/ts_concat.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/ts_concat.hpp"
+
+#include "backend/gna_limitations.hpp"
+#include "common/graph_utils.hpp"
+#include "openvino/cc/ngraph/itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset12.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/transformation_helper.hpp"
+
+using namespace ov;
+using namespace ov::opset12;
+using namespace ov::pass::pattern;
+using namespace ov::intel_gna::graph_utils;
+using namespace ov::intel_gna::pass;
+using namespace ov::intel_gna::pass::helper;
+using namespace ov::intel_gna::limitations;
+
+namespace {
+
+bool is_concat_sinked(const Output<Node>& output) {
+    auto concat_node = ov::as_type_ptr<Concat>(output.get_node_shared_ptr());
+
+    const Shape concat_output_shape = concat_node->get_output_shape(0);
+    const int64_t axis = concat_node->get_concatenation_axis();
+    if (get_first_valuable_dim_id(concat_output_shape) != axis)
+        return false;
+
+    for (size_t i = 0; i < concat_node->get_input_size(); ++i) {
+        auto concat_input = concat_node->input_value(i);
+        auto transpose = ov::as_type_ptr<Transpose>(concat_input.get_node_shared_ptr());
+        if (transpose && !Limitations::is_transpose_supported(transpose))
+            return true;
+    }
+
+    return false;
+}
+
+}  // namespace
+
+TSConcatForward::TSConcatForward() {
+    MATCHER_SCOPE(TSConcatForward);
+
+    auto concat_node_label = wrap_type<Concat>(is_concat_sinked);
+
+    matcher_pass_callback matcher_pass_callback = [=](Matcher& m) {
+        const auto& pattern_to_output = m.get_pattern_value_map();
+
+        auto& concat_node_output = pattern_to_output.at(concat_node_label);
+        auto concat_node = as_type_ptr<Concat>(concat_node_output.get_node_shared_ptr());
+
+        std::vector<AxisVector> gather_indices_vecs;
+        OutputVector concat_inputs;
+        for (size_t i = 0; i < concat_node->get_input_size(); ++i) {
+            auto concat_input = concat_node->input_value(i);
+            auto transpose = ov::as_type_ptr<Transpose>(concat_input.get_node_shared_ptr());
+            const bool is_transposed_input = transpose && !Limitations::is_transpose_supported(transpose);
+
```

Comment:
No need for ov::

Reply:
Discussed on the meeting. We will create separate PR for this transformation